pkg/resource/firewall/hook.go

```diff
@@ -27,3 +31,32 @@ var (
 		ackrequeue.DefaultRequeueAfterDuration,
 	)
 )
+
+func (r *resource) setAdditionalIdentifiers(identifier *ackv1alpha1.AWSIdentifiers) error {
+	if fwPolicyARN, ok := identifier.AdditionalKeys["firewallPolicyARN"]; ok {
+		r.ko.Spec.FirewallPolicyARN = &fwPolicyARN
+	}
+
+	if numSubnetMappings, ok := identifier.AdditionalKeys["numSubnetMappings"]; ok {
+		num, err := strconv.Atoi(numSubnetMappings)
+		if err != nil {
+			return fmt.Errorf("could convert numSubnetMappings to integer: %w", err)
+		}
+		if num > 0 {
+			for i := range numSubnetMappings {
+				key := "subnetMapping" + fmt.Sprint(i)
+				if subnetMapping, ok := identifier.AdditionalKeys[key]; ok {
+					sm := svcapitypes.SubnetMapping{
+						SubnetID: &subnetMapping,
+					}
+					r.ko.Spec.SubnetMappings = append(r.ko.Spec.SubnetMappings, &sm)
+				}
+			}
```

Comment:
Does this mean that the user should privde a `subnetMapping0` or `subnetMapping1`? Not a crazy user experience :/

Reply:
User needs to provide `numSubnetMappings` to indicate how many subnet mappings are there. and then provide the actual subnet mappings starting from `subnetMapping0`.

Please let me know if we can do something better.

Reply:
I think the best is to introduce `runtime.RawExtension` to a new field called `AdditionnalRawKeys` where users can provide any type and we'll have the responsibility parse it properly